### ISSUE TITLE
Problème avec l'API `observationsMailleAndPoint`

### ISSUE THREAD

**Issue by georgemoon** [PnX-SI/GeoNature-atlas#253]
## Problème

Quand j'essaie de voir une fiche espèce, j'ai une erreur 500 avec l'API : `https://<mondomaine>/atlas/api/observationsMailleAndPoint/<id>`

En inspectant le journal avec la commande `supervisorctl tail atlas` on peut voir l'output suivant :
```
geonatureadmin@sig:~$ sudo supervisorctl tail atlas
raise_from_cause
    reraise(type(exception), exception, tb=exc_tb, cause=cause)
  File "/home/geonatureadmin/atlas/venv/lib/python3.5/site-packages/sqlalchemy/util/compat.py", line 153, in reraise
    raise value.with_traceback(tb)
  File "/home/geonatureadmin/atlas/venv/lib/python3.5/site-packages/sqlalchemy/engine/base.py", line 1244, in _execute_context
    cursor, statement, parameters, context
  File "/home/geonatureadmin/atlas/venv/lib/python3.5/site-packages/sqlalchemy/engine/default.py", line 550, in do_execute
    cursor.execute(statement, parameters)
sqlalchemy.exc.ProgrammingError: (psycopg2.ProgrammingError) ERREUR:  la colonne vm_observations_mailles.annee n'existe pas
LINE 1: ...servations_mailles.id_observation) AS nb_obs, max(atlas.vm_o...
```

## Solution

On peut utiliser le script `https://github.com/PnX-SI/GeoNature-atlas/blob/1.4.1/data/observations_mailles.sql` après on a supprimé la vue matérialisée : `DROP MATERIALIZED VIEW atlas.vm_observations_mailles;`. Après l’exécution du script, il faut exécuter la commande : `GRANT SELECT ON TABLE atlas.vm_observations_mailles TO geonatatlas;`.

## Que faire ?

Si je ne me trompe pas, il semble que le script [update_vm_observations.sql](https://github.com/PnX-SI/GeoNature-atlas/blob/1.4.1/data/update_vm_observations.sql) manque la ligne `date_part('year', dateobs) as annee`, que l'on peut voir dans le script [observations_mailles.sql](https://github.com/PnX-SI/GeoNature-atlas/blob/1.4.1/data/observations_mailles.sql).

**Comment by camillemonchicourt** [comment]
Possible car pas certain qu'on ait bien maintenu le fichier "update_vm_observations.sql" car on ne l'utilise pas.

**Comment by georgemoon** [comment]
Dans ce cas, peut-être le fichier devrait être supprimé ainsi que les références dans les instructions d'installation ?

**Comment by camillemonchicourt** [comment]
Oui, on avait créé ce fichier pour faciliter la mise à jour des vues matérialisées (pour faciliter la vie aux administrateurs de la BDD), mais ça fait dupliquer du code, et forcément avec le temps comme tout code dupliqué...
Donc soit on le supprime, soit on le met à jour.
Encore mieux, on pourrait sortir cette partie de ``atlas.sql`` pour qu'elle ne soit pas dupliquée mais pouvoir la jouer de manière isolée.
